pkg/pub_integration/lib/src/headless_env.dart

```diff
@@ -95,10 +106,81 @@ class HeadlessEnv {
       print('Error: $e');
       clientErrors.add(e);
     });
+
+    _trackedPages.add(page);
     return page;
   }
 
+  /// Gets tracking results of [page] and closes it.
+  Future<void> closePage(Page page) async {
```

Comment:
What if we did:
```dart
class TrackedPage {
  final Page page;
  final HeadlessEnv _env;
  TrackedPage._(this.page, this._env);
  Future<void> close() {
     // Collect coverage and close the page...
  }
}
```
Then we could just return `TrackedPage` from `HeadlessEnv.newPage()`.

We could also subclass `Page` and overwrite `close()`, then we wouldn't have to expose the `TrackedPage` type, as it can just be a private class.

Or possibly, better would be to make a `withPage(FutureOr<T> Function<T>() fn)` function which creates the page, calls `fn`, collects coverage numbers and closes the page. I would suspect we end up writing `testWithPage`, or something like that anyways :)